### PR TITLE
respa_exchange: Implement heuristics for determining the real organizer name

### PR DESCRIPTION
**Pull request by juyrjola**
Sometimes Exchange sets the Organizer element to point to the resource mailbox. In these cases, we need to resort to heuristics to figure out who really made the reservation.

If the event has required attendees, we choose the first attendee as the host. If not, we use the DisplayTo field as the reserver name.

### CODE REVIEW

File: respa_exchange/downloader.py [hunk 1]
```diff
@@ -34,37 +35,48 @@ def _populate_reservation(reservation, ex_resource, item_props):
...
         if organizer.given_name and organizer.surname:
-            name = "%s %s" % (organizer.given_name, organizer.surname)
+            name = '%s %s' % (organizer.given_name, organizer.surname)
         else:
             name = organizer.name
-        reservation.reserver_name = name or ""
+        reservation.reserver_name = name or ''
         reservation.host_name = reservation.reserver_name
+    else:
+        reserver_name = item_props.get('reserver_name', None) or ''
```
Comment (tuomas777) [inline respa_exchange/downloader.py]: Unnecessary `None`.

File: respa_exchange/downloader.py [hunk 2]
```diff
@@ -245,8 +292,11 @@ def fetch_reservation_data(ex_reservation):
     )
     session = ex_reservation.exchange.get_ews_session()
     items = [item for item in gcir.send(session)]
+    if len(items) == 0:
+        return None
+
     assert len(items) == 1, "Exchange returned %d items instead of 1" % (len(items))
-    return str(etree.tostring(items[0], pretty_print=True), encoding='utf8')
+    return items[0]
```
Comment (tuomas777) [inline respa_exchange/downloader.py]: Hmm, cannot find any usages of this function?

File: respa_exchange/downloader.py [hunk 3]
```diff
@@ -319,19 +377,23 @@ def sync_from_exchange(ex_resource, future_days=365, no_op=False):
...
 
-            if not ex_reservation:  # It's a new one!
-                ex_reservation = _create_reservation_from_exchange(item_id, ex_resource, item_props)
-            else:
-                if ex_reservation._change_key != item_id.change_key:
-                    # Things changed, so edit the reservation
-                    _update_reservation_from_exchange(item_id, ex_reservation, ex_resource, item_props)
+    with configure_scope() as scope:
+        scope.remove_extra('item_xml')
+        scope.remove_extra('resource')
```
Comment (tuomas777) [inline respa_exchange/downloader.py]: It probably doesn't matter at all in practice, but when `no_op` is `True`, these won't be removed from the scope.

File: respa_exchange/downloader.py (hunk 2 above)
Comment (juyrjola) [reply respa_exchange/downloader.py]: It is used in some auxiliary scripts that are not stored in the repo.

**Pull request merged**